generics/services/users.js

```diff
@@ -6,411 +6,430 @@
  */
 
 //dependencies
-const request = require('request');
+const request = require("request");
 const userServiceUrl = process.env.USER_SERVICE_URL;
 
-const profile = function ( token,userId = "" ) {
-    return new Promise(async (resolve, reject) => {
-        try {
-            
-            let url = userServiceUrl + constants.endpoints.USER_READ_V5;
-
-            if( userId !== "" ) {
-                url = url + "/" + userId + "?"  + "fields=organisations,roles,locations,declarations,externalIds"
-            }
-
-            const options = {
-                headers : {
-                    "content-type": "application/json",
-                    "x-authenticated-user-token" : token
-                }
-            };
-
-            request.get(url,options,kendraCallback);
-
-            function kendraCallback(err, data) {
-
-                let result = {
-                    success : true
-                };
-
-                if (err) {
-                    result.success = false;
-                    console.log("error occured user read api call :",err)
-                } else {
+const profile = function (token, userId = "") {
+  return new Promise(async (resolve, reject) => {
+    try {
+      let url = userServiceUrl + constants.endpoints.USER_READ_V5;
+
+      if (userId !== "") {
+        url =
+          url +
+          "/" +
+          userId +
+          "?" +
+          "fields=organisations,roles,locations,declarations,externalIds";
+      }
+
+      const options = {
+        headers: {
+          "content-type": "application/json",
+          "x-authenticated-user-token": token,
+        },
+      };
+
+      request.get(url, options, kendraCallback);
+
+      function kendraCallback(err, data) {
+        let result = {
+          success: true,
+        };
+
+        if (err) {
+          result.success = false;
+          console.log("error occured user read api call :", err);
+        } else {
+          let response = JSON.parse(data.body);
+          console.log("response from userRead api call :", response);
+          if (
+            response.responseCode ===
+            httpStatusCode["http_responsecode_ok"].message
+          ) {
+            result["data"] = response.result.response;
+            result.data = _.omit(response.result, [
+              "email",
+              "maskedEmail",
+              "maskedPhone",
+              "recoveryEmail",
+              "phone",
+              "lastName",
```

Comment:
@ankitpws From here and other services don't remove lastName

Reply:
removed lastName from here